### ISSUE TITLE
Consider moving CRDs to separate Go module

### ISSUE THREAD

**Issue by kate-goldenring** [spinkube/spin-operator#329]
If in a separate go module, other projects, such as the `spin-kube` plugin could reference it.

Context: https://github.com/spinkube/spin-plugin-kube/issues/88

**Comment by endocrimes** [comment]
It should be able to reference it already? - transitive dependency conflicts shouldn't be much of an issue between the plugin and the operator

**Comment by kate-goldenring** [comment]
@endocrimes good point. we can import the entire spin-operator module and use the api types.
